src/ai/open_ai.rs

```diff
@@ -40,70 +40,142 @@ pub async fn generate_message(
-You should take care of not violate the semantic commit rules when use emojis.
-You should not include any extra content on your response, only the final commit message ready to be included on a git commit.
-  "#,
+                "You should act as a senior software developer \
+                that will generate beautiful git commit messages following the provided \
+                instructions that would be easy to read. Whenever you receive a Ticket \
+                information, you should use these information to help contextualize the \
+                updates you will receive on git diff. You should follow the provided \
+                instructions to create a message that matches with the user pattern. \
+                You should take care of not violate the provided instructions. You \
+                should not include any extra content on your response, only the final \
+                commit message ready to be included on a git commit.",
             ),
         },
         ChatMessage {
-            role: "user".to_string(),
+            role: "user",
             content: format!(
-                r#"
-This is the git diff:
-```
-{}
-```
-            \n\n"#,
-                payload.diff,
+                "This is the git diff:\n \
+                ```\n\
+                {}\n\
+                ```",
+                diff,
             ),
         },
     ];
-    if payload.instructions.len() > 0 {
+
+    if commit_instructions.len() > 0 {
         messages.push(ChatMessage {
-            role: "user".to_string(),
+            role: "user",
             content: format!(
                 "Follow the following instruction as much as possible:\n* {}",
-                payload.instructions.join("* \n")
+                commit_instructions.join("* \n")
             ),
         })
     }
-    match &payload.ticket {
-        Some(ticket) => messages.push(ChatMessage {
-            role: "user".to_string(),
-            content: format!(
-                "Ticket title: {}\nTicket description: {}\n",
-                ticket.title, ticket.description,
-            ),
-        }),
-        None => {}
+
+    if tickets.len() != 0 {
+        messages.push(ChatMessage {
+            role: "user",
+            content: tickets
+                .iter()
+                .map(|ticket| {
+                    format!(
+                        "Task id: {}\nTask title: {}\nTask description: {}\n",
+                        ticket.id, ticket.title, ticket.description,
+                    )
+                })
+                .collect::<Vec<String>>()
+                .join("\n---\n"),
+        });
     }
-    let json_body = serde_json::json!({
-        "model": model,
-        "temperature": 0.2,
-        "messages": messages
-    });
 
-    let response = client
-        .post(url)
-        .headers(headers)
-        .body(serde_json::to_string(&json_body).unwrap())
-        .send()
-        .await;
+    serde_json::json!(messages)
+}
```

Comment:
The diff is a little bit stranger here but now the function `generate_message` only receives a `serde::Value` which is the massages that we are sending to open ai and we have two new functions to create this `serde::Value` which is `get_prompt_for_pull_request` and `get_prompt_for_commit`!